crates/pallet-domains/src/lib.rs

```diff
@@ -247,23 +240,9 @@ mod pallet {
 
             log::trace!(target: "runtime::domains", "Processing fraud proof: {fraud_proof:?}");
 
-            // Revert the execution chain.
-            let (_, mut to_remove) = ReceiptHead::<T>::get();
-
-            let new_best_number: T::BlockNumber = fraud_proof.parent_number.into();
-            let new_best_hash = BlockHash::<T>::get(new_best_number);
-
-            ReceiptHead::<T>::put((new_best_hash, new_best_number));
-
-            while to_remove > new_best_number {
-                let block_hash = BlockHash::<T>::get(to_remove);
-                for (receipt_hash, _) in <ReceiptVotes<T>>::drain_prefix(block_hash) {
-                    Receipts::<T>::remove(receipt_hash);
-                }
-                to_remove -= One::one();
-            }
-
-            // TODO: slash the executor accordingly.
+            // FIXME: currently core domain fraud proof will also submit into primary chain and
+            // handled by this extrinsic incorrectly
```

Comment:
More detail in https://github.com/subspace/subspace/issues/990

Reply:
I prefer not to introduce incomplete refactoring in pallet-receipts, it can simply focus on the state maintenance of happy path, and the unhappy path (fraud proof related) can be another separate PR..